process/headerCheck/headerSignatureVerify.go

```diff
@@ -321,21 +324,37 @@ func (hsv *HeaderSigVerifier) VerifyHeaderWithProof(header data.HeaderHandler) e
 	return hsv.VerifyHeaderProof(prevProof)
 }
 
-func (hsv *HeaderSigVerifier) getHeaderForProof(proof data.HeaderProofHandler) (data.HeaderHandler, error) {
+func (hsv *HeaderSigVerifier) getHeaderForProofAtTransition(proof data.HeaderProofHandler) (data.HeaderHandler, error) {
 	headerUnit := dataRetriever.GetHeadersDataUnit(proof.GetHeaderShardId())
 	headersStorer, err := hsv.storageService.GetStorer(headerUnit)
 	if err != nil {
 		return nil, err
 	}
 
-	return process.GetHeader(proof.GetHeaderHash(), hsv.headersPool, headersStorer, hsv.marshalizer, proof.GetHeaderShardId())
+	var header data.HeaderHandler
+
+	for {
+		header, err = process.GetHeader(proof.GetHeaderHash(), hsv.headersPool, headersStorer, hsv.marshalizer, proof.GetHeaderShardId())
+		if err == nil {
+			break
+		}
+
+		log.Error("getHeaderForProofAtTransition: failed to get header, will wait and try again",
```

Comment:
perhaps downgrade to log.Warn?

Reply:
I would go for a Debug.
maybe an error if we wait too long for the header.